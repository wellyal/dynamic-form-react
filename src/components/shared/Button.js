import React from 'react';
import styled, {keyframes, css} from 'styled-components';

export const Button = props => {
  return <RegularButton {...props}>Button</RegularButton>;
};

const spin = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

const RegularButton = styled.button`
  width: 100%;
  height: 40px;
  border-radius: 18px;
  cursor: pointer;
  border: 0;
  font-size: 15px;
  background-color: ${props => props.theme.buttonEnabled.background};
  color: ${props => props.theme.buttonEnabled.color};

  &:hover {
    opacity: 0.7;
  }

  ${props =>
    props.disabled &&
    css`
      background-color: ${props => props.theme.buttonDisabled.background};
      cursor: not-allowed;
    `};

  ${props =>
    props.loading &&
    css`
      font-size: 0;
      cursor: not-allowed;

      &:before {
        content: '';
        display: block;
        margin: 0 auto;
        width: 18px;
        height: 18px;
        border: 2px solid #fff;
        border-bottom: 2px solid transparent;
        border-right: 2px solid transparent;
        border-radius: 50%;
        animation: ${spin} 1s linear infinite;
      }
    `};
`;
