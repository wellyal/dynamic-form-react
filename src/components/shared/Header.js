import styled, {css} from 'styled-components';

export const Header = styled.div.attrs({
  color: props => props.color || props.theme.primary,
  size: props => props.size || '15px',
  weight: props => props.weight || 'normal',
  align: props => props.align || 'center'
})`
  color: ${props => props.color};
  font-size: ${props => props.size};
  font-weight: ${props => props.weight};
  text-align: ${props => props.align};
  margin: 20px 0;

  ${props =>
    props.primary &&
    css`
    color: ${props => props.theme.primary};
  `}

  ${props =>
    props.secondary &&
    css`
    color: ${props => props.theme.secondary};
  `};
`;
