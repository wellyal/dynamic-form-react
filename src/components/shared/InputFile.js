import React from 'react';
import styled from 'styled-components';
import {compose, withState, withHandlers} from 'recompose';

const enhancer = compose(
  withState('file', 'setFile', null),
  withState('imagePreview', 'setImagePreview', null),
  withHandlers({
    handleUpload: ({setFile, setImagePreview}) => e => {
      e.preventDefault();

      let reader = new FileReader();
      let file = e.target.files[0];
      if (!file) return null;

      reader.onloadend = () => {
        setFile(file);
        setImagePreview(reader.result);
      };

      reader.readAsDataURL(file);
    }
  })
);

export const InputFile = enhancer(
  ({
    item,
    value,
    handleChange,
    handleBlur,
    errors,
    touched,
    handleUpload,
    imagePreview
  }) => {
    return (
      <div>
        <InputFileWrapper>
          <Input
            id={item.id}
            type={item.type}
            value={value}
            onChange={e => {
              handleUpload(e);
              handleChange(e);
            }}
            onBlur={handleBlur}
          />
        </InputFileWrapper>
        {!!imagePreview && (
          <WrapperThumb>
            <Thumb src={imagePreview} />
          </WrapperThumb>
        )}
      </div>
    );
  }
);

const Input = styled.input`
  font-size: 16px;
`;

const InputFileWrapper = styled.div`
  font-size: 20px;
  color: #ccc;
  border-bottom: 1px solid #ccc;
  display: inline-block;
  height: 35px;
  width: 100%;
  display: flex;
  align-items: center;
  padding: 5px;
  position: relative;
  margin: 12px 0;
`;

const WrapperThumb = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  margin: 12px 0;
`;

const Thumb = styled.img`
  max-width: 200px;
  max-height: 200px;
`;
