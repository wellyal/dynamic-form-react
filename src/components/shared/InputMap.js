import React from 'react';
import {get} from 'lodash';
import {compose, withProps, lifecycle} from 'recompose';
import {
  withScriptjs,
  withGoogleMap,
  GoogleMap,
  Marker
} from 'react-google-maps';
import {SearchBox} from 'react-google-maps/lib/components/places/SearchBox';

export const InputMap = compose(
  withProps({
    googleMapURL:
      'https://maps.googleapis.com/maps/api/js?key=AIzaSyC4R6AN7SmujjPUIGKdyao2Kqitzr1kiRg&v=3.exp&libraries=geometry,drawing,places',
    loadingElement: <div style={{height: `100%`}} />,
    containerElement: <div style={{height: `400px`}} />,
    mapElement: <div style={{height: `100%`}} />
  }),
  lifecycle({
    componentWillMount() {
      const refs = {};

      this.setState({
        bounds: null,
        center: {
          lat: 41.9,
          lng: -87.624
        },
        markers: [],
        onMapMounted: ref => {
          refs.map = ref;
        },
        onBoundsChanged: () => {
          this.setState({
            bounds: refs.map.getBounds(),
            center: refs.map.getCenter()
          });
        },
        onSearchBoxMounted: ref => {
          refs.searchBox = ref;
        },
        onPlacesChanged: () => {
          const places = refs.searchBox.getPlaces();
          const bounds = new window.google.maps.LatLngBounds();

          places.forEach(place => {
            if (place.geometry.viewport) {
              bounds.union(place.geometry.viewport);
            } else {
              bounds.extend(place.geometry.location);
            }
          });
          const nextMarkers = places.map(place => ({
            position: place.geometry.location
          }));
          const nextCenter = get(nextMarkers, '0.position', this.state.center);

          this.setState({
            center: nextCenter,
            markers: nextMarkers
          });
        }
      });
    }
  }),
  withScriptjs,
  withGoogleMap
)(props => (
  <GoogleMap
    ref={props.onMapMounted}
    defaultZoom={15}
    center={props.center}
    onBoundsChanged={props.onBoundsChanged}
  >
    <SearchBox
      ref={props.onSearchBoxMounted}
      bounds={props.bounds}
      controlPosition={window.google.maps.ControlPosition.TOP_LEFT}
      onPlacesChanged={props.onPlacesChanged}
    >
      <input
        id={props.item.id}
        value={props.value}
        onChange={props.handleChange}
        type="text"
        placeholder={props.item.placeholder}
        style={{
          boxSizing: `border-box`,
          border: `1px solid transparent`,
          width: `240px`,
          height: `32px`,
          marginTop: `27px`,
          padding: `0 12px`,
          borderRadius: `3px`,
          boxShadow: `0 2px 6px rgba(0, 0, 0, 0.3)`,
          fontSize: `14px`,
          outline: `none`,
          textOverflow: `ellipses`
        }}
      />
    </SearchBox>
    {props.markers.map((marker, index) => (
      <Marker key={index} position={marker.position} />
    ))}
  </GoogleMap>
));
