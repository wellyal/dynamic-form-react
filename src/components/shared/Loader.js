import React from 'react';
import styled, {keyframes} from 'styled-components';

export const Loader = () => (
  <LoaderWrapper>
    <LoaderContent>
      <Spin />
      <Text>Carregando</Text>
    </LoaderContent>
  </LoaderWrapper>
);

const LoaderWrapper = styled.div`
  display: flex;
  height: calc(100vh - 60px);
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const LoaderContent = styled.div`
  text-align: center;
  width: 80px;
  margin: 0 auto;
`;

const Text = styled.p`
  color: #acacac;
  font-size: 15px;
`;

const spin = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

const Spin = styled.div`
  display: inline-block;
  margin-bottom: 5px;
  border: 2px solid transparent;
  border-top: 2px solid ${props => props.theme.success};
  border-left: 2px solid ${props => props.theme.success};
  border-radius: 50%;
  width: 18px;
  height: 18px;
  animation: ${spin} 1s linear infinite;
`;
