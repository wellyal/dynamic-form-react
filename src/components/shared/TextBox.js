import React from 'react';
import styled from 'styled-components';
import {compose, pure, withProps} from 'recompose';

const enhancer = compose(
  withProps(({errors, item, touched}) => ({
    error: errors[item.id] && touched[item.id],
    success: !errors[item.id] && touched[item.id],
    initial: !touched[item.id]
  })),
  pure
);

export const TextBox = enhancer(
  ({
    item,
    value,
    handleChange,
    handleBlur,
    errors,
    touched,
    error,
    success,
    initial
  }) => {
    const status = {error, success, initial};
    return (
      <TextBoxWrapper {...status}>
        <Input
          id={item.id}
          required
          type={item.type}
          value={value}
          onChange={handleChange}
          onBlur={handleBlur}
        />
        <Label {...status} htmlFor={[item.type]}>
          {item.placeholder}
        </Label>
      </TextBoxWrapper>
    );
  }
);

const Label = styled.label.attrs({
  color: props => {
    if (props.initial) return props.theme.secondary;
    else if (props.error) return props.theme.danger;
    else if (props.success) return props.theme.success;
  }
})`
  font-size: 15px;
  position: absolute;
  top: 22px;
  transition: all 0.2s;
  color: ${props => props.color};
`;

const Input = styled.input.attrs({
  color: props => {
    if (props.initial) return props.theme.secondary;
    else if (props.error) return props.theme.danger;
    else if (props.success) return props.theme.success;
  }
})`
  border: 0;
  width: 100%;
  height: 20px;
  position: relative;
  outline: 0;
  font-size: 20px;
  top: 10px;
  z-index: 19;
  background: transparent;

  &:focus ~ ${Label}, &:active ~ ${Label}, &:valid ~ ${Label} {
    position: absolute;
    top: 7px;
    font-size: 11px;
    transition: all 0.2s;
    color: ${props => props.color};
  }
`;

const TextBoxWrapper = styled.div.attrs({
  color: props => {
    if (props.initial) return props.theme.secondary;
    else if (props.error) return props.theme.danger;
    else if (props.success) return props.theme.success;
  }
})`
  font-size: 20px;
  color: #ccc;
  border-bottom: 1px solid ${props => props.color};
  display: inline-block;
  height: 35px;
  width: 100%;
  display: flex;
  align-items: center;
  padding: 5px;
  position: relative;
  margin: 12px 0;
`;
