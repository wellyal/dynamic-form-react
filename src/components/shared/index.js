export * from './Button';
export * from './TextBox';
export * from './Header';
export * from './Wrapper';
export * from './InputFile';
export * from './InputMap';
export * from './ListView';
export * from './Loader';
