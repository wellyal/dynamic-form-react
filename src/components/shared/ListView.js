import React from 'react';
import styled from 'styled-components';
import {compose, pure, branch, renderComponent} from 'recompose';

import {Header} from '~/components/shared';

const enhancer = compose(
  branch(
    ({registersList}) => !registersList,
    renderComponent(() => <div>Nenhum cadastro efetuado</div>)
  ),
  pure
);

export const ListView = enhancer(({registersList}) => {
  console.log('registersList', registersList);
  return (
    <div>
      <Header size={20}>Listagem</Header>
      {registersList.map(item => {
        return (
          <Row key={item.txtCPF}>
            <div>{item.txtFullname}</div>
            <div>{item.txtPhone}</div>
          </Row>
        );
      })}
    </div>
  );
});

const Row = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  padding: 10px 0;
  border-bottom: 1px solid #ccc;
`;
