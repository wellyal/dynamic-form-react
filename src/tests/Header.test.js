import React from 'react'
import styled from 'styled-components'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

// locals
import {Header} from '~/components/shared'
import {theme} from '~/config'

test('should have default color Header', () => {
  const tree = renderer.create(<Header theme={theme} />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('color', '#333')
})

test('should have default color size', () => {
  const tree = renderer.create(<Header theme={theme} />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-size', '15px')
})

test('should have default weight', () => {
  const tree = renderer.create(<Header theme={theme} />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-weight', 'normal')
})

test('should have default Header align', () => {
  const tree = renderer.create(<Header theme={theme} />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('text-align', 'center')
})

test('should replace Header color', () => {
  const tree = renderer.create(<Header theme={theme} color="#555"/>).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('color', '#555')
})

test('should replace Header size', () => {
  const tree = renderer.create(<Header theme={theme} size="50px"/>).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-size', '50px')
})

test('should replace Header weight', () => {
  const tree = renderer.create(<Header theme={theme} weight="bold"/>).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-weight', 'bold')
})

test('should replace Header color', () => {
  const tree = renderer.create(<Header theme={theme} align="left"/>).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('text-align', 'left')
})
