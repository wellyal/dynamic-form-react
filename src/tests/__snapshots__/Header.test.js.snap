// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have default Header align 1`] = `
.c0 {
  color: #333;
  font-size: 15px;
  font-weight: normal;
  text-align: center;
  margin: 20px 0;
}

<div
  align="center"
  className="c0"
  color="#333"
  size="15px"
  weight="normal"
/>
`;

exports[`should have default color Header 1`] = `
.c0 {
  color: #333;
  font-size: 15px;
  font-weight: normal;
  text-align: center;
  margin: 20px 0;
}

<div
  align="center"
  className="c0"
  color="#333"
  size="15px"
  weight="normal"
/>
`;

exports[`should have default color size 1`] = `
.c0 {
  color: #333;
  font-size: 15px;
  font-weight: normal;
  text-align: center;
  margin: 20px 0;
}

<div
  align="center"
  className="c0"
  color="#333"
  size="15px"
  weight="normal"
/>
`;

exports[`should have default weight 1`] = `
.c0 {
  color: #333;
  font-size: 15px;
  font-weight: normal;
  text-align: center;
  margin: 20px 0;
}

<div
  align="center"
  className="c0"
  color="#333"
  size="15px"
  weight="normal"
/>
`;

exports[`should replace Header color 1`] = `
.c0 {
  color: #555;
  font-size: 15px;
  font-weight: normal;
  text-align: center;
  margin: 20px 0;
}

<div
  align="center"
  className="c0"
  color="#555"
  size="15px"
  weight="normal"
/>
`;

exports[`should replace Header color 2`] = `
.c0 {
  color: #333;
  font-size: 15px;
  font-weight: normal;
  text-align: left;
  margin: 20px 0;
}

<div
  align="left"
  className="c0"
  color="#333"
  size="15px"
  weight="normal"
/>
`;

exports[`should replace Header size 1`] = `
.c0 {
  color: #333;
  font-size: 50px;
  font-weight: normal;
  text-align: center;
  margin: 20px 0;
}

<div
  align="center"
  className="c0"
  color="#333"
  size="50px"
  weight="normal"
/>
`;

exports[`should replace Header weight 1`] = `
.c0 {
  color: #333;
  font-size: 15px;
  font-weight: bold;
  text-align: center;
  margin: 20px 0;
}

<div
  align="center"
  className="c0"
  color="#333"
  size="15px"
  weight="bold"
/>
`;
